WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEmailFragment.java

```diff
@@ -0,0 +1,397 @@
+        if (actionBar != null) {
+            actionBar.setDisplayShowTitleEnabled(false);
+            actionBar.setDisplayHomeAsUpEnabled(true);
+        }
+
+        if (savedInstanceState != null) {
+            boolean isInProgress = savedInstanceState.getBoolean(KEY_IN_PROGRESS);
+            if (isInProgress) {
+                showEmailCheckProgressDialog();
+            }
+        }
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (context instanceof LoginStateGetter.FsmGetter) {
+            mLoginNavInputEmail = ((LoginStateGetter.FsmGetter) context).getLoginStateGetter().getLoginNavInputEmail();
+        } else {
+            throw new RuntimeException(context.toString() + " must implement LoginStateGetter.FsmGetter");
+        }
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        mLoginNavInputEmail = null;
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+
+        outState.putBoolean(KEY_IN_PROGRESS, mProgressDialog != null);
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        inflater.inflate(R.menu.menu_login, menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (item.getItemId() == R.id.help) {
+            mLoginNavInputEmail.help();
+            return true;
+        }
+
+        return false;
+    }
+
+    /*
+     * autofill the username and password from BuildConfig/gradle.properties (developer feature,
+     * only enabled for DEBUG releases)
+     */
+    private void autofillFromBuildConfig() {
+        if (!BuildConfig.DEBUG) return;
+
+        String email = (String) WordPress.getBuildConfigValue(getActivity().getApplication(),
+                "DEBUG_DOTCOM_LOGIN_EMAIL");
+        if (!TextUtils.isEmpty(email)) {
+            mEmailEditText.setText(email);
+            AppLog.d(T.NUX, "Autofilled email from build config");
+        }
+    }
+
+    private void autocorrectEmail() {
+        if (mEmailAutoCorrected) {
+            return;
+        }
+        final String email = EditTextUtils.getText(mEmailEditText).trim();
+        // Check if the username looks like an email address
+        final Pattern emailRegExPattern = Patterns.EMAIL_ADDRESS;
+        Matcher matcher = emailRegExPattern.matcher(email);
+        if (!matcher.find()) {
+            return;
+        }
+        // It looks like an email address, then try to correct it
+        String suggest = EmailChecker.suggestDomainCorrection(email);
+        if (suggest.compareTo(email) != 0) {
+            mEmailAutoCorrected = true;
+            mEmailEditText.setText(suggest);
+            mEmailEditText.setSelection(suggest.length());
+        }
+    }
+
+    protected void onDoneAction() {
```

Comment:
Good catch. Another relic from the port from the SignInFragment.

Took the opportunity to clean this up some more, inheriting directly from Fragment instead of AbstractFragment. Fixed with 2ed752b.